nx/test/nx/defn/evaluator_test.exs

```diff
@@ -32,6 +32,13 @@ defmodule Nx.Defn.EvaluatorTest do
     assert %T{shape: {3, 2}, type: {:s, 64}} = reshape(Nx.iota({2, 3}))
   end
 
+  defn reduce_window(t1, acc),
+    do: Nx.window_reduce(t1, acc, {2}, [padding: :valid], fn x, acc -> x + acc end)
+
+  test "window reduce" do
+    assert reduce_window(Nx.tensor([1, 2, 3]), 0) == Nx.tensor([3, 5])
```

Comment:
So the problem is actually on the accumulator being passed as a number, right?

It seems to me that this should also break outside of BinaryBackend.

Perhaps the "to_tensor" call should actually be in the Nx module itself instead of the backend implementation.

Reply:
Also, if this is the case and my suggestion does fix the problem, the BinaryBackend changes should be rolled back

Reply:
> So the problem is actually on the accumulator being passed as a number, right?

`x` value also is passed as a number. And pre-converting acc to scalar before actually starting reduce process
https://github.com/elixir-nx/nx/blob/ce396667e54ee7096d9ab05506df083949fbda84/nx/lib/nx/binary_backend.ex#L1361
will break some non-defn cases.

Reply:
Thanks for the explanation!

Reply:
And for the PR :D